modules/delivery/delivery.service.js

```diff
@@ -0,0 +1,244 @@
+const request = require('request');
+const util = require('util')
+const post = util.promisify(request.post)
+const { 
+    CITY_NOT_FOUND,
+    WAREHOUSE_NOT_FOUND,
+ } = require('../../error-messages/delivery.message')
+
+class NovaPoshtaService {
+  async getNovaPoshtaCities(city){  
+      const res = await post(
+          process.env.NOVA_POSHTA_API_LINK,
+          { 
+            json: true,
+            body: {
+              modelName: 'Address',
+              calledMethod: 'getCities',
+              methodProperties: {
+                  FindByString: city
+              },
+              apiKey: process.env.NOVA_POSHTA_API_KEY
+            }
+          },
+      );
+
+      if(res.body.errors.lenght) {
+          throw new Error(CITY_NOT_FOUND)
+      }
+      return res.body.data.slice(0, 10);
+  }
+
+  async getNovaPoshtaStreet(CityRef, street) {
+      const res = await post(
+          process.env.NOVA_POSHTA_API_LINK,
+          { 
+            json: true,
+            body: {
+              modelName: 'Address',
+              calledMethod: 'getStreet',
+              methodProperties: {
+                  CityRef,
+                  FindByString: street
+              },
+              apiKey: process.env.NOVA_POSHTA_API_KEY
+            }
+          },
+      );
+
+      return res.body.data.slice(0, 10)
+  }
+
+  async getNovaPoshtaWarehouses(city){  
+      const res = await post(
+          process.env.NOVA_POSHTA_API_LINK,
+          { 
+            json: true,
+            body: {
+              modelName: 'AddressGeneral',
+              calledMethod: 'getWarehouses',
+              methodProperties: {
+                  CityName: city
+              },
+              apiKey: process.env.NOVA_POSHTA_API_KEY
+            }
+          },
+      );
+
+      if(res.body.errors.lenght) {
```

Comment:
Please, lowercase the name of the argument, it is a plain value, not a type itself

Reply:
fixed